### PR TITLE
Add german contributing guidelines

### PR DESCRIPTION
**Pull request by lupinitylabs**
## What does this PR do?
Add a German version of the contribution guidelines.

### Description
Complementing the work on https://github.com/EbookFoundation/free-programming-books/pull/5684, I added a translation for the contribution guidelines in German and added links to all other contribution files' `Read this in other languages` lines.

Instead of appending the link to the end of the list, I re-ordered the first part of the list alphabetically. Let me know if there's a reason why it wasn't alphabetical before, and I will change it accordingly.

### Why is this valuable (or not)?

It improves accessibility for the community beyond native english speakers.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.

## Follow-up

- [x] Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: CONTRIBUTING-de.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    - *Interaktive Tutorials* : Eine interaktive Webseite, die den Benutzer Sourcecode oder Kommandos eingeben lässt und das Resultat auswertet (mit "auswerten" meinen wir nicht "bewerten"). z.B.: [Try Haskell](http://tryhaskell.org), [Try Github](http://try.github.io).
+    - *Podcasts und Screencasts* : Podcasts und Screencasts.
+    - *Problem Sets & Competitive Programming* : Eine Webseite oder Software, die dir die Möglichkeit gibt, deine Programmierfähigkeiten durch die Lösung einfacher oder komplexer Problemstellungen auf die Probe zu stellen, mit oder ohne Code Review und mit oder ohne den Vergleich der Leistungen mit anderen Besuchern der Seite.
+
+4. Stell sicher, dass du den [Richtlinien](#richtlinien) folgst und die [Markdown Formatierung](#formatierung) der Dateien beachtest.
+
+5. GitHub Actions werden Tests ausführen, um sicherzustellen, dass die Listen korrekt alphabetisiert sind and den Formatierungsregeln Folge geleistet wurde. Stell sicher, dass deine Änderungen diese Tests bestehen.
```
Comment (davorpa) [inline CONTRIBUTING-de.md:23]: It satisfy #5644 :tada:

File: CONTRIBUTING-de.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+Kurse kommen oftmals mit begleitenden Lehrbüchern, die wir in unseren Bücherlisten führen würden. Kurse bieten Vorträge, Übungen, Tests, Anmerkungen oder andere Lernhilfen. Ein einzelner Vortrag oder Video allein ist kein Kurs. Eine Powerpoint-Präsentation ist kein Kurs.  
+
+##### Interaktive Tutorials vs. anderes Zeug
+
+Wenn etwas ausgedruckt werden kann, ohne dass es seinen Nutzen verliert, ist es kein interaktives Tutorial.  
+
+### Automatisierung
+
+- Die Durchsetzung der Formatierungsregeln wird über [GitHub Actions](https://github.com/features/actions) mittels [fpb-lint](https://github.com/vhf/free-programming-books-lint) sichergestellt (siehe [.github/workflows/fpb-lint.yml](.github/workflows/fpb-lint.yml))
```
Comment (davorpa) [inline CONTRIBUTING-de.md:203]: It satisfy #5644 :tada:

**Review by davorpa — CHANGES_REQUESTED** [review]
> Instead of appending the link to the end of the list, I re-ordered the first part of the list alphabetically. Let me know if there's a reason why it wasn't alphabetical before, and I will change it accordingly.

**I am for the work of sticking to what is strictly necessary**. It's simple: **avoid anoying merge conflicts** due to have so many PRs touching the first paragraph. We can always deal with it when all of them will be commited.

**:question: Could you swap back French and Spanish?**

**Comment by lupinitylabs** [comment]
Sure, will do. Where should the German link go then at the end, or is there any other concept?

**Comment by lupinitylabs** [comment]
I swapped back French and Spanish. If you want me to put German at the end or anywhere else, please let me know. 👍

**Comment by davorpa** [comment]
> Sure, will do. Where should the German link go then - at the end, or is there any other concept?

At the moment as it is. To handle at #5762

File: CONTRIBUTING-fa_IR.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-*این متن را در زبان‌های دیگر بخوانید: [English](CONTRIBUTING.md), [Français](CONTRIBUTING-fr.md), [Español](CONTRIBUTING-es.md), [简体中文](CONTRIBUTING-zh.md), [繁體中文](CONTRIBUTING-zh_TW.md), [Tiếng Việt](CONTRIBUTING-vn.md), [Русский](CONTRIBUTING-ru.md).*
+*این متن را در زبان‌های دیگر بخوانید: [Deutsch](CONTRIBUTING-de.md), [English](CONTRIBUTING.md) , [Français](CONTRIBUTING-fr.md), [Español](CONTRIBUTING-es.md), [简体中文](CONTRIBUTING-zh.md), [繁體中文](CONTRIBUTING-zh_TW.md), [Tiếng Việt](CONTRIBUTING-vn.md), [Русский](CONTRIBUTING-ru.md).*
```
Comment (davorpa) [inline CONTRIBUTING-fa_IR.md:1]: There are some extra space after English and it comma

**Comment by lupinitylabs** [comment]
> 
> 
> > Sure, will do. Where should the German link go then - at the end, or is there any other concept?
> 
> At the moment as it is. To handle at #5762

Nice 👍 You sure have an issue for just about everything 😲

**Review by davorpa — CHANGES_REQUESTED** [review]

File: CONTRIBUTING-fa_IR.md (hunk 3 above)
Comment (lupinitylabs) [reply CONTRIBUTING-fa_IR.md]: Sorry about that, it slipped my attention... RTL can be a little awkward to edit 😅

Comment (davorpa) [reply CONTRIBUTING-fa_IR.md]: I agree. And mixing glyph types is a bit weird due to, at least for me, don't know where put the cursor 😥

**Review by davorpa — APPROVED** [review]
thanks for your valuable effort 🙄

### Ready to merge :fireworks: :tada:

**Comment by eshellman** [comment]
I agree that there should be some order to the language lists, but now is a bad time to do it because of conflicts. Could you add an issue about that so it gets taken care of? 
Thanks, and welcome to free-programming-books!

**Comment by lupinitylabs** [comment]
> I agree that there should be some order to the language lists, but now is a bad time to do it because of conflicts. Could you add an issue about that so it gets taken care of? Thanks, and welcome to free-programming-books!

Yes, agreed. An issue already exists at https://github.com/EbookFoundation/free-programming-books/issues/5762 as @davorpa pointed out earlier.

Glad to contribute. Thanks for your work on the repo!

**Pull request merged**